### PR TITLE
Add API endpoints to get Category Tags.

### PR DESCRIPTION
**Pull request by vlvereta**
Some frontend work at the start but then backend support work on getting tags.

### CODE REVIEW

File: app/Actions/Place/GetPlaceTagsByCategory/GetPlaceTagsByCategoryAction.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public function __construct(PlaceCategoryTagRepository $repository)
+    {
+        $this->categoryTagsRepository = $repository;
+    }
+
+    public function execute(GetPlaceTagsByCategoryRequest $request): GetPlaceTagsByCategoryResponse
+    {
+        $tags = $this->categoryTagsRepository->findByCriteria(
+            new PlaceCategoryTagCriteria($request->getCategoryId())
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceTagsByCategory/GetPlaceTagsByCategoryAction.php]: Easier probably to add categoryTagsRepository->findByCategory(int $categoryId) method and use `whereHas` inside

File: routes/api.php [hunk 2]
```diff
@@ -131,8 +131,8 @@
             'create', 'edit'
         ]);
 
-        Route::post('/places/categories/search', 'Api\Place\PlaceCategoryController@getPlaceCategoryByName');
+        Route::get('/places/categories/{id}/tags', 'Api\Place\PlaceCategoryTagsController@getTagsByCategoryId');
 
-        /* Routes here.. */
+        Route::post('/places/categories/search', 'Api\Place\PlaceCategoryController@getPlaceCategoryByName');
```
Comment (nemoipaha) [inline routes/api.php]: this is `Route::get()`.

File: tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            'place_tag_id'      => $tag->id
+        ]);
+
+        $this->assertDatabaseHas('place_category_place_tag',
+            [
+                'place_category_id' => $category->id,
+                'place_tag_id'      => $tag->id
+            ]
+        );
+
```
Comment (nemoipaha) [inline tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php]: empty line

File: tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    public function testGetTagsNoCategory()
+    {
+        $categoryId = 99999999;
+
+        $this->assertDatabaseMissing('place_categories',
+            [
+                'id' => $categoryId
+            ]
+        );
+
```
Comment (nemoipaha) [inline tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php]: empty line

File: tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        $category = PlaceCategory::create([
+            'name' => 'testCategory'
+        ]);
+
+        $this->assertDatabaseHas('place_categories',
+            [
+                'id' => $category->id,
+            ]
+        );
+
```
Comment (nemoipaha) [inline tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php]: same)

File: frontend/src/store/modules/place/category/actions.js [hunk 6]
```diff
@@ -13,5 +13,17 @@ export default {
                     reject(err);
                 });
         });
+    },
+
+    getAllCategories: () => {
+        return new Promise((resolve, reject) => {
+            httpService.get('/places/categories')
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/category/actions.js]: Result will be the same if use
```
return httpService.get('/places/categories')
                .then(function (result) {
                    resolve(result.data.data);
                })
```

File: frontend/src/pages/NewPlacePage.vue [hunk 7]
```diff
@@ -377,14 +382,49 @@ export default {
...
         },
 
         isDaySelected: function () {
             return !this.weekdays.length;
         }
+    },
+
+    methods: {
+        onCloseTab: function (tagObject) {
+            this.newPlace.category_tags = this.newPlace.category_tags.filter((tag) => tag.name !== tagObject.name);
```
Comment (nemoipaha) [inline frontend/src/pages/NewPlacePage.vue]: `this.newPlace.category_tags.filter((tag) => tag.name !== tagObject.name);` should be separate method

File: frontend/src/pages/NewPlacePage.vue [hunk 8]
```diff
@@ -377,14 +382,49 @@ export default {
...
+                // this.$store.dispatch('placeCategoryTags/getTagsByCategoryId', categoryObject.id)
+                //     .then((result) => {
+                //         this.category_tags = result;
+                //     });
+                this.selectedTag = 'v';
+            }
+        },
+
+        'selectedTag': function (tagObject) {
+            if (tagObject !== 'v' && !this.newPlace.category_tags.some((tag) => tag.name === tagObject.name)) {
```
Comment (nemoipaha) [inline frontend/src/pages/NewPlacePage.vue]: another method `this.newPlace.category_tags.some((tag) => tag.name === tagObject.name)`

File: frontend/src/pages/NewPlacePage.vue [hunk 9]
```diff
@@ -377,14 +382,49 @@ export default {
...
+                this.newPlace.category_tags.push(tagObject);
+                console.log(this.selectedTag.name);
```
Comment (nemoipaha) [inline frontend/src/pages/NewPlacePage.vue]: remove)

File: frontend/src/pages/NewPlacePage.vue [hunk 10]
```diff
@@ -377,14 +382,49 @@ export default {
...
+            .then((result) => {
+                this.categories = result;
+            });
+    },
+    
+    watch: {
+        'newPlace.category': function (categoryObject) {
+            if (categoryObject) {
+                this.newPlace.category_tags = [];
+                // this.$store.dispatch('placeCategoryTags/getTagsByCategoryId', categoryObject.id)
```
Comment (nemoipaha) [inline frontend/src/pages/NewPlacePage.vue]: remove comments

File: frontend/src/pages/NewPlacePage.vue [hunk 11]
```diff
@@ -13,7 +13,12 @@
                                 <div class="field-body">
                                     <div class="field">
                                         <div class="control is-expanded">
```
Comment (nemoipaha) [inline frontend/src/pages/NewPlacePage.vue]: NewPlacePage.vue is wrapper component. Create `AddPlaceForm.vue` and move code there

File: routes/api.php (hunk 2 above)
Comment (vlvereta) [reply routes/api.php]: Not mine part of code.
There is POST 'cause  this route passing data with name of the category to set it in action here -> https://github.com/BinaryStudioAcademy/bsa-2018-hedonist/blob/468ef4ffaf7bc2d63f18b75115d3fc6f0f43386b/frontend/src/store/modules/place/category/actions.js#L7

File: app/Actions/Place/GetPlaceTagsByCategory/GetPlaceTagsByCategoryAction.php (hunk 1 above)
Comment (vlvereta) [reply app/Actions/Place/GetPlaceTagsByCategory/GetPlaceTagsByCategoryAction.php]: Done.

File: frontend/src/pages/NewPlacePage.vue (hunk 11 above)
Comment (vlvereta) [reply frontend/src/pages/NewPlacePage.vue]: Done.

File: frontend/src/pages/NewPlacePage.vue (hunk 10 above)
Comment (vlvereta) [reply frontend/src/pages/NewPlacePage.vue]: Done.

File: frontend/src/pages/NewPlacePage.vue (hunk 9 above)
Comment (vlvereta) [reply frontend/src/pages/NewPlacePage.vue]: Done.

File: frontend/src/store/modules/place/category/actions.js (hunk 6 above)
Comment (vlvereta) [reply frontend/src/store/modules/place/category/actions.js]: Does not work.

File: app/Actions/Place/GetPlaceTagsByCategory/GetPlaceTagsByCategoryAction.php (hunk 1 above)
Comment (vlvereta) [reply app/Actions/Place/GetPlaceTagsByCategory/GetPlaceTagsByCategoryAction.php]: I think so too, but I thought that as we have `findByCriteria` method we should use it for more complicated requests.
I changed logic and add `findByCategory` method into `PlaceCategoryTagRepository` 'cause it's common repository for tags but `categoryTagsRepository` is for places tags in use, as I understand.

File: tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php (hunk 3 above)
Comment (vlvereta) [reply tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php]: I leave it for more readable code structure)
Removed.

File: tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php (hunk 4 above)
Comment (vlvereta) [reply tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php]: Removed.

File: tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php (hunk 5 above)
Comment (vlvereta) [reply tests/Feature/Api/Place/PlaceCategoryTagsControllerTest.php]: Removed.

**Pull request merged**